partner_auto_archive/views/res_partner_views.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_partner_form_auto_archive" model="ir.ui.view">
+        <field name="name">res.partner.form.auto_archive</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form" />
+        <field name="arch" type="xml">
+            <!-- Add checkbox to set as auto-archive -->
+            <xpath expr="//span[@name='address_name']" position="before">
+                <field name="auto_archive" />
+            </xpath>
+            <xpath expr="//label[@for='street']" position="before">
+                <field name="auto_archive" />
+            </xpath>
```

Comment:
maybe it would be better to put the field after "category_id" so that it is shown in the right column of field information